### ISSUE TITLE
AKS Cluster VMSS cannot be queried via Get-AzureRMVMSS or Get-AzVMSS or Get-AzResource or Get-AzureRMResource

### ISSUE THREAD

**Issue by jmos5156** [Azure/AKS#1730]
**What happened**:
After deploying a private AKS cluster with Linux and Windows node pools the VMSS's cannot be enumerated using the Powershell Cmdlets `get-azurermvmss` or `get-azvmss`. They do not return the list of scale sets from the AKS cluster.

The same applies to running `get-azurermresource` and `get-azresource`. The AKS vmss do not appear in the lists.

The command only works if you specify both the resource group and vmss names as parameters. This is not optimal if you want to query and filter for specific scale sets or list all in the subscription.

However, running `az list vmss -o table` does work.

**What you expected to happen**:
I expect to be able to use the above cmdlets to return a complete list of all the VMSS a subscription.

**How to reproduce it (as minimally and precisely as possible)**:
Create a VMSS or AKS cluster and on completion run the above commands.

**Anything else we need to know?**:
The AKS cluster is provisioned via ARM template and completes successfully. On occasions, if we wait ~ 30 mins, the cmdlets do sometimes return what we expect.
I am running the latest versions of the AzureRM and Az Modules.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16.10
- Size of cluster (how many worker nodes are in the cluster?) 3 pools each with 2 nodes
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.)
- Others:

**Comment by palma21** [comment]
Please reach out to the Powershell repo as that is not AKS related (https://github.com/Azure/azure-powershell). Do you have the latest modules (up to date)?

Nonetheless is there a reason you want to get the underlying VMSS resource? As the regular cautionary advice bear in mind AKS does not support any operations done at the underlying IaaS level.

**Comment by alexeldeib** [comment]
Seems surprising, are you sure you're using the same subscription? AKS VMSS are no different than other VMSS, unless .NET SDK/PowerShell have some bizarre caching I'm unaware of that experience is quite surprising. 

Definitely check with powershell folks

**Comment by jmos5156** [comment]
Thanks for the responses. We can replicate the issue on AzureRM module version 5.7.0 and 6.13.1. We now have a support ticket with Azure and they confirm the same. This could be more of a timing issue with Azure metadata refresh. We have found that by leaving a newly deploy private windows AKS cluster for some time, the cmdlets will work, however, the timespan is not concrete (can be mins or hours in some cases).

The reason for this effort is to enable system assigned identity to the windows node pool (vmss) to grant pods access to keyvault. This issue has already been addressed with our Azure AKS CSA, who confirms that the current workaround is to manually enable SI on the VMSS. This is addressed in another github issue. We are, however, trying to automate this part as we have no clear date for a fix.

I will raise this in the Powershell github repo.

Many Thanks
